### ISSUE TITLE
consistency in validation.yml location

### ISSUE THREAD

**Issue by mariev** [phuse-org/valtools#81]
It appears that the expected location of `validation.yml` changes with validation model.  

within an R package e.g. "validate on install" 
```
- package_dir
   |- R
   |- vignettes  <<-  working_dir/output_dir
   |   |- validation
   |   |   |- requirements
   |   |   |- test code
   |   |   |- test cases
   |   |- validation_report.Rmd
   |   |- ...
   |- validation.yml
   |- ...
```

vs "validate after install " 

```
- package_dir
   |- R
   |- inst <<-  working_dir/output_dir
   |   |- validation
   |   |   |- requirements
   |   |   |- test code
   |   |   |- test cases
   |   |   |- validation_report.Rmd
   |   |   |- validation.yml
   |   |- ...
   | ...
```

vs "validate outside of package"

```
 |- user_dir <<- working_dir/output_dir
   |- validation
   |   |- requirements
   |   |- test code
   |   |- test cases
   |   |- validation_report.Rmd
   |   |- validation.yml
   | ...
```   

Should we move towards a consistent framework where the file `validation.yml` lives in the location identified as "working_dir". So  `vignettes`, `inst`, or user-defined by respective models? 

If we retain the current variation, I could use a snippet of code that will capture the relative path change to `validation.yml` from `validation_report.Rmd` and works even after copying files e.g. from `vignettes/` to `inst/`

**Comment by mariev** [comment]
wrt to capturing the path to `validation.yml`, perhaps a modification to `find_file` could help?

Current:
```r
> find_file("validation.yml", package_file())
[1] "validation.yml"
```

Proposed:
```r
> find_file("validation.yml", package_file(), full.names = TRUE)
[1] "C:/Users/mvendett/AppData/Local/Temp/RtmpABRgJ7/validation.yml"
```

**Comment by mariev** [comment]
Per today's discussion, move `validation.yml` to live in `validation` folder for all modes

**Comment by thebioengineer** [comment]
want me to change the `vt_use_validation_config()` function? also, would it be wise to just add this as a child of `vt_use_validation()`? always felt a little weird to me to have these two functions with similar names live next to one another

**Comment by mariev** [comment]
🤔 good point. Would renaming `vt_use_validation_config` -->  `vt_use_config`  gain some distance?

**Comment by thebioengineer** [comment]
I think it vt_use_config might make sense to call vt_use_validation, or make them one function. They are intrinsicly reliant on one another it seems like once we move the validation.yml into the "working dir":
- `vt_use_config`: creates validation.yml in the specified "working directory",
- `vt_use_validation`: creates the working directory identifed by the config. 

@mariev @elimillera

**Comment by mariev** [comment]
🤔 I recall some conversations that we wanted to leave the door open for users not comfortable/interested in maintaining a config yaml. In the last PHUSE mtg, Phil mentioned that some users may see prep of a yaml to be an insurmountable barrier 😬. Would `vt_use_validation` continue to provide non-config based access?

**Comment by thebioengineer** [comment]
What I took from Phil is that they wouldn't want to maintain/interact with the yml manually. We are providing tools to interact with it such that the user functionally doesn't know it exists. @philbowsher - Care to weigh in?

**Comment by philbowsher** [comment]
Spot on - I really like YAML but I teach a lot of workshops and it is rarely known about by attendees. I think it's an awesome tool and as more and more people in the pharma space come to know how software development works, YAML will be a big part of it I suspect.

Even in RMD, via the YAML header, many people in my workshops don't know it's YAML. There are packages like ymlthis, but the more this process can find a way to support the current knowledge by the users is a good thing in my opinion.

Really just depends on the audience and who will be using the process. If it is more advanced users, and those in software dev, then it could be fine.

**Comment by thebioengineer** [comment]
I have most of the code together now. working through the tests now. 

going forward, the location of contents will be all referenceable from locations of the validation.yml:

within an R package e.g. "validate on install" 
```
- package_dir
   |- R
   |- vignettes  <<-  working_dir/output_dir
   |   |- validation
   |   |   |- requirements
   |   |   |- test code
   |   |   |- test cases
   |   |   |- validation.yml
   |   |- validation_report.Rmd
   |   |- ...
   |- ...
```

vs "validate after install "  (this also gets constructed when a package using the `validate on install` if `valtools::vt_validate_install()` is used to install the package)

```
- package_dir
   |- R
   |- inst <<-  working_dir/output_dir
   |   |- validation
   |   |   |- requirements
   |   |   |- test code
   |   |   |- test cases
   |   |   |- validation_report.Rmd
   |   |   |- validation.yml
   |   |- ...
   | ...
```

vs "validate outside of package"

```
 |- user_dir <<- working_dir/output_dir
   |- validation
   |   |- requirements
   |   |- test code
   |   |- test cases
   |   |- validation_report.Rmd
   |   |- validation.yml
   | ...
```    

question: for validate outside of package, should validation_report.Rmd be moved up a level? Otherwise things are just nested within a validation folder . what would be the purpose of a validation folder specifically?

**Comment by mariev** [comment]
I was under the impression that output_dir would be where the pdf file would be rendered to?? So it would be possible to have the portion suitable for sigs located in `user_dir` and all the "stuff" of making things work live in `validation`. Not wholly committed to that idea, though!

**Comment by thebioengineer** [comment]
resolved with #82
